### PR TITLE
Add MangoAccount and Health Calculations

### PR DESCRIPTION
**Pull request by bdhobare**
#13 
- Added [`MangoAccount`](https://github.com/blockworks-foundation/mango-client-v3/blob/main/src/MangoAccount.ts).
- Added [`serum::OpenOrders`](https://github.com/project-serum/serum-dex/blob/master/dex/src/state.rs#L587).
- Added [Health calculations](https://github.com/blockworks-foundation/mango-client-v3/blob/main/src/MangoAccount.ts#L932) from the Typscript SDK.
- Ported [Unit tests](https://github.com/blockworks-foundation/mango-client-v3/pull/50) from the Typescript SDK.
- Added example to read account health.

### CODE REVIEW

**Comment by mschneider** [comment]
a lot of the calculations convert to I80F48 way too early. double has only 64bits of precision and will introduce a lot of rounding errors.

**Comment by bdhobare** [comment]
Replaced the fixed implementation with the one from [cpp-utilities](https://github.com/eteran/cpp-utilities/blob/master/fixed/include/cpp-utilities/fixed.h) and remove premature conversions to double from i80f48.

File: include/utils.hpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    return std::make_tuple(mangoGroup.spotMarkets[marketIndex].initAssetWeight,
+                           mangoGroup.spotMarkets[marketIndex].initLiabWeight,
+                           mangoGroup.perpMarkets[marketIndex].initAssetWeight,
+                           mangoGroup.perpMarkets[marketIndex].initLiabWeight);
+  } else {
+    return std::make_tuple(i80f48(1.0), i80f48(1.0), i80f48(1.0), i80f48(1.0));
+  }
+}
+auto nativeI80F48ToUi(i80f48 amount, uint8_t decimals) {
+  return amount / pow(10, decimals);
```
Comment (mschneider) [inline include/utils.hpp:56]: which pow is this and does it have enough precision? 12 decimals is the largest i have seen

Comment (mschneider) [reply include/utils.hpp:56]: i think using 10LL instead would be more correct

File: tests/main.cpp [hunk 2]
```diff
@@ -137,4 +137,497 @@ TEST_CASE("Test getMultipleAccounts") {
...
+  auto mangoGroup =
+      solana::rpc::fromFile<mango_v3::MangoGroup>(path + "/group.json");
+  auto accountInfo =
+      solana::rpc::fromFile<mango_v3::MangoAccountInfo>(path + "/account.json");
+  auto mangoAccount = mango_v3::MangoAccount(accountInfo);
+  auto mangoCache =
+      solana::rpc::fromFile<mango_v3::MangoCache>(path + "/cache.json");
+  auto initHealth = mangoAccount.getHealth(mangoGroup, mangoCache,
+                                           mango_v3::HealthType::Init);
+  CHECK_EQ(initHealth.to_double(), 37904260000.059059);
```
Comment (mschneider) [inline tests/main.cpp:182]: how did you generate these results?

File: include/utils.hpp (hunk 1 above)
Comment (mschneider) [reply include/utils.hpp:56]: just looked up pow, it's apparently using float/double under the hood, so 10.0L would be correct

File: tests/main.cpp (hunk 2 above)
Comment (bdhobare) [reply tests/main.cpp:182]: Hey @mschneider these values are copied from the TS SDK. Specifically this PR: https://github.com/blockworks-foundation/mango-client-v3/pull/50

**Review by mschneider — APPROVED** [review]

**Pull request merged**